precompiles/staking/events.go

```diff
@@ -122,6 +124,36 @@ func (p Precompile) EmitAllowanceChangeEvent(ctx sdk.Context, stateDB vm.StateDB
 	return nil
 }
 
+// EmitCreateValidatorEvent creates a new ctreate valdator event emitted on a CreateValidator transaction.
+func (p Precompile) EmitCreateValidatorEvent(ctx sdk.Context, stateDB vm.StateDB, msg *stakingtypes.MsgCreateValidator, delegatorAddr common.Address) error {
+	// Prepare the event topics
+	event := p.ABI.Events[EventTypeCreateValidator]
+	topics, err := p.createStakingTxTopics(3, event, delegatorAddr, msg.ValidatorAddress)
+	if err != nil {
+		return err
+	}
+
+	// Prepare the event data
+	var b bytes.Buffer
+	b.Write(cmn.PackNum(reflect.ValueOf(msg.Commission.Rate.BigInt())))
+	b.Write(cmn.PackNum(reflect.ValueOf(msg.Commission.MaxRate.BigInt())))
+	b.Write(cmn.PackNum(reflect.ValueOf(msg.Commission.MaxChangeRate.BigInt())))
+	b.Write(cmn.PackNum(reflect.ValueOf(msg.MinSelfDelegation.BigInt())))
+	// TODO: I emit the CreateValidator event in the contract by deploy a normal contract. Determine the encoding of Value and Pubkey
+	b.Write(common.Hex2Bytes("00000000000000000000000000000000000000000000000000000000000000c0"))
+	b.Write(cmn.PackNum(reflect.ValueOf(msg.Value.Amount.BigInt())))
+	b.Write(cmn.PackElement(reflect.ValueOf(msg.Pubkey.String())))
+
+	stateDB.AddLog(&ethtypes.Log{
+		Address:     p.Address(),
+		Topics:      topics,
+		Data:        b.Bytes(),
+		BlockNumber: uint64(ctx.BlockHeight()),
```

Comment:
I generally try to avoid using manually packing like this unless there are 1-2 arguments. You should use something like this instead:
```go
	// Pack the arguments to be used as the Data field
	commission := msg.Commission
	arguments := abi.Arguments{
		event.Inputs[2],
		event.Inputs[3],
		event.Inputs[4],
		.... // Add more here
	}
	packed, err := arguments.Pack(
		commission.Rate.BigInt(),
		commission.MaxRate.BigInt(),
		commission.MaxChangeRate.BigInt(),
		... // Add more here
	)
	if err != nil {
		return err
	}
```

Reply:
yes please

Reply:
done !